Failure to build 4.5.1 with CMake

I’m trying to build the latest release using CMake (3.15.3), but it fails:
```
    cmake \
        -DCMAKE_BUILD_TYPE=Release \
        -DENABLE_FILTER_TEST=ON \
        -DLARGE_FILE_TESTS=ON \
        -DTEST_WITH_VALGRIND=ON
```
```
-- The Fortran compiler identification is GNU 9.1.0
-- The C compiler identification is GNU 9.1.0
-- Check for working Fortran compiler: /usr/bin/f95
-- Check for working Fortran compiler: /usr/bin/f95  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/f95 supports Fortran 90
-- Checking whether /usr/bin/f95 supports Fortran 90 -- yes
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.16") found components:  doxygen missing components:  dot
-- Found m4: /usr/bin/m4
-- Found netcdf library: /usr/lib/libnetcdf.so
-- Found netcdf.h in /usr/include
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for malloc.h
-- Looking for malloc.h - found
-- Looking for ctype.h
-- Looking for ctype.h - found
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for errno.h
-- Looking for errno.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Looking for stdbool.h
-- Looking for stdbool.h - found
-- Looking for locale.h
-- Looking for locale.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdio.h
-- Looking for stdio.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for stdarg.h
-- Looking for stdarg.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for sys/dir.h
-- Looking for sys/dir.h - found
-- Looking for sys/ndir.h
-- Looking for sys/ndir.h - not found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for pstdint.h
-- Looking for pstdint.h - not found
-- Looking for endian.h
-- Looking for endian.h - found
-- Looking for BaseTsd.h
-- Looking for BaseTsd.h - not found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netcdf.h
-- Looking for netcdf.h - found
-- Check size of void*
-- Check size of void* - done
-- Check size of char
-- Check size of char - done
-- Check size of double
-- Check size of double - done
-- Check size of float
-- Check size of float - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of long long
-- Check size of long long - done
-- Check size of off_t
-- Check size of off_t - done
-- Check size of off64_t
-- Check size of off64_t - failed
-- Check size of short
-- Check size of short - done
-- Check size of size_t
-- Check size of size_t - done
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Check size of __int64
-- Check size of __int64 - failed
-- Check size of uchar
-- Check size of uchar - failed
-- Check size of int64_t
-- Check size of int64_t - done
-- Check size of uint64_t
-- Check size of uint64_t - done
-- Looking for alloca
-- Looking for alloca - not found
-- Looking for nc_def_opaque in /usr/lib/libnetcdf.so
-- Looking for nc_def_opaque in /usr/lib/libnetcdf.so - found
-- Looking for nccreate in /usr/lib/libnetcdf.so
-- Looking for nccreate in /usr/lib/libnetcdf.so - found
-- Looking for nc_set_log_level in /usr/lib/libnetcdf.so
-- Looking for nc_set_log_level in /usr/lib/libnetcdf.so - found
-- Looking for oc_open in /usr/lib/libnetcdf.so
-- Looking for oc_open in /usr/lib/libnetcdf.so - found
-- Check size of signed char
-- Check size of signed char - done
Matching Fortran types to C types
Matching Fortran types to C types - done


-- CMake Summary:

-- 	Build Type:			Release
-- 	Building Shared Libraries:	ON
-- 	Building netCDF-4:		ON
-- 	Building DAP Support:		ON
-- 	Logging Support:		ON
-- 	Enabling CDF5:			TRUE
-- 	Relaxed check NC_EINVALCOORDS:	TRUE
-- 	Building Parallel IO:		OFF
-- 	Using NetCDF4 Parallel IO:	FALSE
-- 	Using PnetCDF Parallel IO:	FALSE
-- 	Testing Parallel IO:		OFF
-- 	Linking against:		/usr/lib/libnetcdf.so;

-- Configuring done
CMake Error at nf03_test4/CMakeLists.txt:88 (ADD_EXECUTABLE):
  Cannot find source file:

    tst_flarge.f90

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at nf03_test4/CMakeLists.txt:88 (ADD_EXECUTABLE):
  No SOURCES given to target: tst_flarge


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -395,6 +395,7 @@ MACRO(build_bin_test F ext)
 ENDMACRO()
 
 OPTION(ENABLE_LARGE_FILE_TESTS "Run large file tests, which are slow and take lots of disk." OFF)
+SET(LARGE_FILE_TESTS ${ENABLE_LARGE_FILE_TESTS})
 OPTION(BUILD_BENCHMARKS "Run F90 I/O Benchmarks" OFF)
 OPTION(TEST_WITH_VALGRIND "Run extra tests with valgrind" OFF)
 OPTION(ENABLE_PARALLEL_TESTS "Run parallel I/O tests for F90 and F77" OFF)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -394,7 +394,7 @@ MACRO(build_bin_test F ext)
   ENDIF()
 ENDMACRO()
 
-OPTION(LARGE_FILE_TESTS "Run large file tests, which are slow and take lots of disk." OFF)
+OPTION(ENABLE_LARGE_FILE_TESTS "Run large file tests, which are slow and take lots of disk." OFF)
 OPTION(BUILD_BENCHMARKS "Run F90 I/O Benchmarks" OFF)
 OPTION(TEST_WITH_VALGRIND "Run extra tests with valgrind" OFF)
 OPTION(ENABLE_PARALLEL_TESTS "Run parallel I/O tests for F90 and F77" OFF)

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -6,7 +6,13 @@ Release Notes {#nf_release_notes}
 This file contains a high-level description of this package's evolution.
 Entries are in reverse chronological order (most recent first).
 
-## 4.5.2 - TBD
+## 4.5.2 -
+
+### Requirements
+
+### Changes
+
+* Corrected an issue where cmake-specific large file tests weren't being captured by `make dist`.
 
 ## 4.5.1 - September 4, 2019
 

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -12,7 +12,7 @@ Entries are in reverse chronological order (most recent first).
 
 ### Changes
 
-* Corrected an issue where cmake-specific large file tests weren't being captured by `make dist`.
+* Corrected an issue where cmake-specific large file tests weren't being captured by `make dist`. See [Github #198](https://github.com/Unidata/netcdf-fortran/issues/198) for more details.
 
 ## 4.5.1 - September 4, 2019
 

--- a/nf03_test4/CMakeLists.txt
+++ b/nf03_test4/CMakeLists.txt
@@ -49,10 +49,10 @@ if (USE_NETCDF4)
 
   # This is the netCDF-4 F90 large file test.
   IF (LARGE_FILE_TESTS)
-    SET(nc4_largefile_PROGRAMS tst_flarge)
+    SET(nc4_largefile_PROGRAMS f90tst_flarge)
     SET(check_PROGRAMS ${check_PROGRAMS} ${nc4_largefile_PROGRAMS})
     SET(TESTS ${TESTS} ${nc4_largefile_PROGRAMS})
-    SET(tst_flarge_SOURCES tst_flarge.f90)
+    SET(f90tst_flarge_SOURCES f90tst_flarge.f90)
   ENDIF(LARGE_FILE_TESTS)
 
   # This is an f90 benchmark.

--- a/nf03_test4/Makefile.am
+++ b/nf03_test4/Makefile.am
@@ -34,11 +34,11 @@ f90tst_types_SOURCES = f90tst_types.f90
 f90tst_types2_SOURCES = f90tst_types2.f90
 
 # This is the netCDF-4 F90 large file test.
-if ENABLE_LARGE_FILE_TESTS
+if LARGE_FILE_TESTS
 check_PROGRAMS += f90tst_flarge
 TESTS += f90tst_flarge
 f90tst_flarge_SOURCES = f90tst_flarge.f90
-endif # ENABLE_LARGE_FILE_TESTS
+endif # LARGE_FILE_TESTS
 
 # This is an f90 benchmark.
 if BUILD_BENCHMARKS

--- a/nf03_test4/Makefile.am
+++ b/nf03_test4/Makefile.am
@@ -12,7 +12,7 @@ AM_FCFLAGS = -I$(top_builddir)/fortran
 # All tests need to link to fortran library.
 LDADD = ${top_builddir}/fortran/libnetcdff.la
 
-# tst_f90_nc4 
+# tst_f90_nc4
 NC4_F90_TESTS = f90tst_vars f90tst_vars_vlen f90tst_grps f90tst_fill	\
 f90tst_fill2 f90tst_vars2 f90tst_vars3 f90tst_vars4 f90tst_path		\
 f90tst_rengrps f90tst_nc4 f90tst_types f90tst_types2
@@ -34,11 +34,11 @@ f90tst_types_SOURCES = f90tst_types.f90
 f90tst_types2_SOURCES = f90tst_types2.f90
 
 # This is the netCDF-4 F90 large file test.
-if LARGE_FILE_TESTS
+if ENABLE_LARGE_FILE_TESTS
 check_PROGRAMS += f90tst_flarge
 TESTS += f90tst_flarge
 f90tst_flarge_SOURCES = f90tst_flarge.f90
-endif # LARGE_FILE_TESTS
+endif # ENABLE_LARGE_FILE_TESTS
 
 # This is an f90 benchmark.
 if BUILD_BENCHMARKS
@@ -65,7 +65,7 @@ endif # TEST_PARALLEL
 # check-valgrind target, which runs all tests with valgrind.
 @VALGRIND_CHECK_RULES@
 
-EXTRA_DIST = CMakeLists.txt run_f90_par_test.sh
+EXTRA_DIST = CMakeLists.txt run_f90_par_test.sh f90tst_flarge.f90
 
 # Cleaning up files created during the testing.
 CLEANFILES = f90tst_*.nc fort.*

